packages/server/src/rooms.ts

```diff
@@ -64,3 +64,30 @@ async function doJoinRoom(io: IOServer, socket: Socket, store: Store, room: Room
     io.in(room.name).emit(GameEvent.CurrentState, state);
   }
 }
+
+export function updateGameState(io: IOServer, store: Store, roomName: string, gameState: GameState) {
+  store.updateRoomState(roomName, gameState);
+  io.in(roomName).emit(GameEvent.CurrentState, gameState);
+  const isRoundFinshed = gameState.players.every((player) => player.hand.length === 0);
+
+  const isMatchFinished = isRoundFinshed && gameState.table.length === 0;
+
+  if (isMatchFinished) {
+    // TODO send the final score need to figure out who was the last player to capture cards because
+    // all the final cards on the table will need to go to them
+    io.in(roomName).emit(GameStatus.Ended);
+  } else if (isRoundFinshed) {
+    const { deck, players, ...rest } = gameState;
+    const updatedPlayersHands = players.map((player) => ({
+      ...player,
+      hand: deck.splice(0, 3),
+    }));
+    const updatedGameState = {
+      ...rest,
+      players: updatedPlayersHands,
+      deck,
+    };
+    store.updateRoomState(roomName, updatedGameState);
```

Comment:
Hm... Is the UI going to add some ordered animations, where you first see the opponent playing, and then the new turn happening?

Reply:
yeah I have to think how to do this, basically we have to add a timeout everytime there is a state update and run some animations before the state is actually applied, quite a pain, not sure how is going to work or what's the best solution

Reply:
Phaser uses a concept of _tweens_. A tween is a scheduled movement of an object, that can move from point A to B, with various interpolation ways (Linear, Ease, etc), You can have them being delayed, be run in a loop, and also queue them together. So, I think we need a way to queue animations, that reference the same object (for example a card can move from point A to B, and then from where it was to C).